core/src/apps/monero/xmr/crypto/__init__.py

```diff
@@ -37,9 +37,19 @@ def keccak_2hash(inp, buff=None):
     return buff
 
 
-def compute_hmac(key, msg=None):
-    h = hmac.new(key, msg=msg, digestmod=keccak_factory)
-    return h.digest()
+def compute_hmac(key, msg):
+    digestmod = keccak_factory
+    inner = digestmod()
+    block_size = inner.block_size
+    if len(key) > block_size:
+        key = digestmod(key).digest()
+    key = key + bytes(block_size - len(key))
+    inner.update(bytes((x ^ 0x36) for x in key))
+    inner.update(msg)
+    outer = digestmod()
+    outer.update(bytes((x ^ 0x5C) for x in key))
+    outer.update(inner.digest())
+    return outer.digest()
```

Comment:
the `bytes(bla for bla in foo)` are huge in terms of allocation count
something like this is better (untested):

```suggestion
def compute_hmac(key, msg):
    digestmod = keccak_factory
    inner = digestmod()
    block_size = inner.block_size
    if len(key) > block_size:
        key = digestmod(key).digest()
    key_block = bytearray(block_size)
    for i in range(block_size):
        key_block[i] = 0x36
    for i in range(len(key)):
    	key_block[i] ^= key[i]
    inner.update(key_block)
    inner.update(msg)
    outer = digestmod()
    for i in range(block_size):
        key_block[i] = 0x5C
    for i in range(len(key)):
    	key_block[i] ^= key[i]
    outer.update(key_block)
    outer.update(inner.digest())
    return outer.digest()
```